<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DialogConstraint</class>
 <widget class="QDialog" name="DialogConstraint">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>700</width>
    <height>490</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QWidget{
	background:transparent;
}

QLabel{
	color: rgb(0,0,127);
}

QDialog{
	background-image: url(:/backgrounds/yande.re 268248.jpg);
}

QTabWidget{
	border: 1px solid white;
}

QTabWidget::pane{
	border: 1px solid white;
	padding: 0 0 0 0px;
}

QFrame#frameNormal, QFrame#frameChain{
	background-color:rgba(255,255,255,160);
}

QTabBar::tab {
	background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:0, y2:1, stop:0 rgba(255, 255, 255, 64), stop:1 rgba(255, 255, 255, 127));
}

QTabBar::tab:selected {
	background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:0, y2:1, stop:0 rgba(255, 255, 255, 64), stop:1 rgba(255, 255, 255, 255));
}

QTabBar::tab:hover {
	background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:0, y2:1, stop:0 rgba(255, 255, 255, 64), stop:1 rgba(255, 255, 255, 255));
}

QGroupBox{
	border: 1px solid white;
	margin-top: 12px;
}

QGroupBox::title {
    subcontrol-origin: margin;
    subcontrol-position: top left;
	background-color: rgba(255, 255, 255, 160);
	border-color:white;
	color: rgb(0,0,127);
}

QLineEdit{
	color: rgb(0,0,127);
	background-color: rgba(255, 255, 255, 127);
	border: 1px solid white;
	border-radius: 2px;
}

QLineEdit::hover{
	background-color: rgba(255, 255, 255, 192);
}

QSpinBox,QDoubleSpinBox {
	color: rgb(0,0,127);
	background-color: rgba(255, 255, 255, 127);
	border: 1px solid white;
	border-radius: 2px;
}

QSpinBox:hover,QDoubleSpinBox:hover {
	background-color: rgba(255, 255, 255, 192);
}

QSpinBox::up-button,QDoubleSpinBox::up-button,
QSpinBox::down-button,QDoubleSpinBox::down-button {
	border-left: 1px solid white;
	background-color: transparent;
}

QSpinBox::up-button:hover,
QDoubleSpinBox::up-button:hover,
QSpinBox::down-button:hover,
QDoubleSpinBox::down-button:hover
{
	border-left: 1px solid white;
	background-color: rgba(255, 255, 255, 127);
}

QSpinBox::up-arrow,QDoubleSpinBox::up-arrow{
	image: url(:/icons/arrowup.png);
	width: 5px;
}

QSpinBox::down-arrow,QDoubleSpinBox::down-arrow{
	image: url(:/icons/arrowdown.png);
	width: 5px;
}

QPushButton{
	color: rgb(0,0,127);
	background-color: rgba(255, 255, 255, 127);
	border: 1px solid white;
	border-radius: 5px;
	padding: 4 20 4 20px;
}

QToolButton{
	color: rgb(0,0,127);
	background-color: rgba(255, 255, 255, 127);
	border: 1px solid white;
	border-radius: 5px;
}

QToolButton::menu-button{
	border-left: 1px solid white;
}

QToolButton::menu-arrow{
	image: url(:/icons/arrowdown.png);
	width: 3;
}

QPushButton:hover, QToolButton:hover{
	background-color: rgba(255, 255, 255, 192);
}

QPushButton:pressed, QToolButton:pressed{
	background-color: rgba(255, 255, 255, 224);
}

QComboBox{
	color: rgb(0,0,127);
	background-color: rgba(255, 255, 255, 127);
	border: 1px solid white;
	border-radius: 2px;
}

QComboBox:hover{
	background-color: rgba(255, 255, 255, 192);
}

QComboBox:on{
	background-color: rgba(255, 255, 255, 192);
}

QComboBox::drop-down{
	border-left:1px solid white;
}

QComboBox::drop-down::hover{
	border-left:1px solid white;
	background-color: rgba(255, 255, 255, 192);
}

QComboBox::down-arrow{
	image: url(:/icons/arrowdown.png);
	width: 7px;
}

QComboBox::down-arrow{
	image: url(:/icons/arrowdown.png);
	width: 7px;
}

QComboBoxListView{
	background-color: white;
}

QListWidget{
	border:1px solid white;
	border-radius:2px;
}

QMenu{
	color: rgb(0,0,127);
	background-color:white;
	border: 1px solid white;
}</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="1" column="0">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tabNormal">
      <attribute name="title">
       <string>Normal</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <widget class="QFrame" name="frameNormal">
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>9</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QFrame" name="frameType">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="label_4">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Type:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboType">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QFrame" name="frameBodies">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QHBoxLayout" name="layoutBodies">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="label">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Body A:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboBodyA">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_2">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>Body B:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="comboBodyB">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QGroupBox" name="groupBoxData">
            <property name="sizePolicy">
             <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="title">
             <string>Constraints Settings:</string>
            </property>
            <layout class="QGridLayout" name="layoutData"/>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tabChain">
      <attribute name="title">
       <string>Chain</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item row="0" column="0">
        <widget class="QFrame" name="frameChain">
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QGridLayout" name="gridLayout_5">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="0" column="1" rowspan="3">
           <widget class="QFrame" name="frameData">
            <property name="sizePolicy">
             <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QComboBox" name="comboChainType"/>
             </item>
             <item>
              <widget class="QGroupBox" name="groupBox">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="title">
                <string>Common Setting:</string>
               </property>
               <layout class="QFormLayout" name="formLayout">
                <item row="0" column="0">
                 <widget class="QLabel" name="label_5">
                  <property name="text">
                   <string>Tau:</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="label_6">
                  <property name="text">
                   <string>Damping:</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QDoubleSpinBox" name="chainTau">
                  <property name="decimals">
                   <number>6</number>
                  </property>
                  <property name="maximum">
                   <double>1.000000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>0.010000000000000</double>
                  </property>
                  <property name="value">
                   <double>0.800000000000000</double>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QDoubleSpinBox" name="chainDamping">
                  <property name="decimals">
                   <number>6</number>
                  </property>
                  <property name="maximum">
                   <double>1.000000000000000</double>
                  </property>
                  <property name="singleStep">
                   <double>0.010000000000000</double>
                  </property>
                  <property name="value">
                   <double>0.800000000000000</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QGroupBox" name="groupBox_2">
               <property name="title">
                <string>Node Setting:</string>
               </property>
               <layout class="QGridLayout" name="gridLayout_6">
                <item row="0" column="0">
                 <layout class="QGridLayout" name="layoutChainData"/>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QListWidget" name="listWidget">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="editTriggers">
             <set>QAbstractItemView::NoEditTriggers</set>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_3">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Rigidbodies:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QFrame" name="horizontalFrame">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QComboBox" name="comboChainBody"/>
             </item>
             <item>
              <widget class="QToolButton" name="buttonAdd">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>+</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QToolButton" name="buttonRemove">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>-</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QToolButton" name="buttonUp">
               <property name="text">
                <string>...</string>
               </property>
               <property name="arrowType">
                <enum>Qt::UpArrow</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QToolButton" name="toolButton">
               <property name="text">
                <string>...</string>
               </property>
               <property name="arrowType">
                <enum>Qt::DownArrow</enum>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>DialogConstraint</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>257</x>
     <y>390</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>DialogConstraint</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>325</x>
     <y>390</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
